apps/src/templates/maker/EligibilityConfirmDialog.jsx

```diff
@@ -0,0 +1,148 @@
+/** @file Confirm Dialog for Maker Discount Codes */
+import React, {Component, PropTypes} from 'react';
+import i18n from "@cdo/locale";
+import BaseDialog from "../BaseDialog";
+import DialogFooter from "../teacherDashboard/DialogFooter";
+import Button from "../Button";
+
+const styles = {
+  subtitle: {
+    fontSize: 16,
+  },
+  form: {
+    margin: '10px 0',
+  },
+  signature: {
+    margin: '5px 0'
+  },
+  signatureBox: {
+    height: 40,
+    width: '100%',
+    boxSizing: 'border-box',
+    padding: 10,
+  },
+  checkboxes: {
+    paddingLeft: 5,
+    marginBottom: 20
+  },
+  checkboxLabel: {
+    marginLeft: 25,
+  },
+  checkbox: {
+    marginLeft: -25,
+    width: 25,
+  },
+};
+
+export default class EligibilityConfirmDialog extends Component {
+  static propTypes = {
+    onCancel: PropTypes.func.isRequired,
+    onSuccess: PropTypes.func.isRequired,
+  };
+
+  state = {
+    signature: "",
+    validInput: false,
+    submitting: false,
+  };
+
+  verifyResponse = () => {
+    this.setState({
+      validInput: this.check1.checked
+        && this.check2.checked
+        && this.check3.checked
+        && /\S/.test(this.state.signature)
+    });
+  }
+
+  handleSubmit = () => {
+    this.setState({submitting: true});
+    // TODO: extract actual signature and make an ajax POST
+    // fake async for now
+    setTimeout(() => {
+      this.setState({
+        submitting: false,
+      });
+      const discountCode = '12345';
+      this.props.onSuccess(discountCode);
+    }, 1000);
+  }
+
+  setSignature = (event) => {
+    this.setState({signature: event.target.value}, this.verifyResponse);
+  }
+
+  render() {
+    return (
+      <BaseDialog
+        useUpdatedStyles
+        uncloseable
+        isOpen
+        assetUrl={() => ''}
```

Comment:
I really don't like that this is a required prop for BaseDialog. I might make a separate PR to fix this up.

Reply:
I second this motion.